api/process_test.go

```diff
@@ -131,3 +144,96 @@ func TestProcess(t *testing.T) {
+	adminToken := testCreateUser(t, "adminpassword123")
+
+	// Create org and census
+	orgAddress, censusID, _ := testCreateOrgAndCensus(t, adminToken)
+
+	// Generate a random process ID
+	processID := internal.HexBytes(util.RandomBytes(32))
+	t.Logf("Generated process ID for draft test: %s\n", processID.String())
+
+	censusIDBytes := internal.HexBytes{}
+	err := censusIDBytes.ParseString(censusID)
+	c.Assert(err, qt.IsNil)
+
+	// Step 1: Create a process with draft=true
+	initialMetadata := map[string]any{
+		"title":       "Draft Process",
+		"description": "This is a draft process",
+	}
+	draftProcessInfo := &apicommon.CreateProcessRequest{
+		OrgAddress: orgAddress,
+		CensusID:   censusIDBytes,
+		Metadata:   initialMetadata,
+		Draft:      true, // Mark as draft
+	}
+
+	pid := requestAndParseWithAssertCode[string](
+		http.StatusOK,
+		t,
+		http.MethodPost,
+		adminToken,
+		draftProcessInfo,
+		"process",
+	)
+	t.Log("Successfully created draft process")
+
+	// Verify the process was created and is in draft mode
+	resp, code := testRequest(t, http.MethodGet, adminToken, nil, "process", pid)
+	c.Assert(code, qt.Equals, http.StatusOK, qt.Commentf("response: %s", resp))
+
+	var createdProcess db.Process
+	err = parseJSON(resp, &createdProcess)
+	c.Assert(err, qt.IsNil)
+	c.Assert(createdProcess.Draft, qt.IsTrue, qt.Commentf("Process should be in draft mode"))
+	t.Log("Verified process is in draft mode")
+
+	// Verify the list of draft processes contains 1 item
+	{
+		processDraftsResp := requestAndParse[apicommon.ListOrganizationProcesses](t,
+			http.MethodGet, adminToken, nil, "organizations", orgAddress.String(), "processes", "drafts")
+		c.Assert(processDraftsResp.Processes, qt.HasLen, 1)
+	}
+
+	// Step 2: Update the process with new metadata and draft=false
+	updatedMetadata := map[string]any{
+		"title":       "Updated Process",
+		"description": "This is no longer a draft process",
+	}
+	draftFalse := false
+	updatedProcessInfo := &apicommon.UpdateProcessRequest{
+		CensusID: censusIDBytes,
+		Metadata: updatedMetadata,
+		Draft:    &draftFalse, // No longer a draft
+	}
+
+	resp, code = testRequest(t, http.MethodPut, adminToken, updatedProcessInfo, "process", pid)
+	c.Assert(code, qt.Equals, http.StatusOK, qt.Commentf("response: %s", resp))
+	t.Log("Successfully updated process and set draft=false")
+
+	// Verify the process was updated and is no longer in draft mode
+	resp, code = testRequest(t, http.MethodGet, adminToken, nil, "process", pid)
+	c.Assert(code, qt.Equals, http.StatusOK, qt.Commentf("response: %s", resp))
+
+	var updatedProcess db.Process
+	err = parseJSON(resp, &updatedProcess)
+	c.Assert(err, qt.IsNil)
+	c.Assert(updatedProcess.Draft, qt.IsFalse, qt.Commentf("Process should no longer be in draft mode"))
+	c.Assert(updatedProcess.Metadata, qt.DeepEquals, updatedMetadata, qt.Commentf("Process metadata should be updated"))
+	t.Log("Verified process is no longer in draft mode and metadata was updated")
+
```

Comment:
i had a test here `// Step 3: Try to update the process again, which should fail since it's no longer in draft mode` and it's gone, so is this design change on purpose?

Reply:
i reintroduced the test